spec/factories.rb

```diff
@@ -282,6 +310,43 @@
     description "Description for factory-generated General question."
     max_grade 1
     staff_comments "Staff comment for factory-generated General question."
+
+    trait :auto_grading_exact do
+      auto_graded true
+      auto_grading_type :exact
+      ignore do
+        option_count 2
+      end
+      after(:build) do |question, evaluator|
+        evaluator.option_count.times do
+          question.auto_grading_exact_options << build(:auto_grading_exact_option)
+        end
+      end
+    end
+
+    trait :auto_grading_keyword do
+      auto_graded true
+      auto_grading_type :keyword
+      ignore do
+        option_count 2
+      end
+      after(:build) do |question, evaluator|
+        evaluator.option_count.times do
+          question.auto_grading_keyword_options << build(:auto_grading_keyword_option)
+        end
+      end
+    end
+  end
+
+  factory :auto_grading_exact_option, class: Assessment::AutoGradingExactOption do
+    correct true
+    answer 'Some answer'
+    explanation 'Some explanation'
+  end
+
+  factory :auto_grading_keyword_option, class: Assessment::AutoGradingKeywordOption do
+    keyword 'keyword1'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.